packages/api/tests/models/test-collections-model.js

```diff
@@ -100,13 +84,14 @@ test('Collection.delete() throws an exception if the collection has associated r
 test('Collection.delete() deletes a collection', async (t) => {
   const name = randomString();
   const version = randomString();
+  const cache = collectionsModel.collectionConfigStore.cache;
+  const initialCacheSize = Object.keys(cache).length;
 
-  await manager.create(fakeCollectionFactory({ name, version }));
-
-  t.true(await manager.exists({ name, version }));
+  await collectionsModel.create(fakeCollectionFactory({ name, version }));
+  t.true(await collectionsModel.exists(name, version));
+  t.is(Object.keys(cache).length, initialCacheSize + 1);
```

Comment:
I think we probably also want to test that the record is written to S3 on create and then removed from S3 on delete?

Reply:
That would require whitebox testing, meaning that the test would have to use knowledge of the implementation details of the config store. Specifically, the test would have to know where and how the object is stored in S3 in order to confirm it's existence (or absence).

In fact, this change to the test is also a whitebox test because it uses knowledge of the cache, which after further thought seems to me like something that we shouldn't do either. Further, that's really more a test of the config store rather than of the collection model.

Unfortunately, there's currently no other way to readily confirm that the model's delete method does now indeed delete the entry from the config store. That's also unfortunately due to the fact that the config store is not a dependency that is injected into the model, so there's no way to pass in my own config store to confirm that it's delete method is invoked when the model's delete method is invoked.

Another possibility would be to define an `exists` method on the config store and call through to it to confirm that it returns `false` after calling the model's `delete` method.

Let me know what you would prefer to do here, if not just leave this as is.

Reply:
Agreed on all points. I'm in favor of injecting the config store as a dependency and adding an `exists` method to it, which we should add a separate unit test for in the tests for that class.

Reply:
I'll implement the dependency injection and make use of it in the tests. Shall I create a Jira ticket for adding an `exists` method to the config store class?

Reply:
I don't think we need a separate ticket - I think adding that method is an implementation detail for testing the work for this ticket

Reply:
Actually, I'm not going to implement `CollectionConfigStore.exists` because I can use the `get` method to determine whether or not it exists. If it doesn't exist, `get` will throw an exception, so I'm going to use that behavior in the test to confirm that `Collection.delete` successfully called through to `CollectionConfigStore.delete`.